pkg/proxy/handlers.go

```diff
@@ -159,6 +165,11 @@ func (p *Proxy) withImpersonateRequest(handler http.Handler) http.Handler {
 
 // newErrorHandler returns a handler failed requests.
 func (p *Proxy) newErrorHandler() func(rw http.ResponseWriter, r *http.Request, err error) {
+	unauthedHandler := audit.NewUnauthenticatedHandler(p.auditor, func(rw http.ResponseWriter, r *http.Request) {
+		klog.V(2).Infof("unauthenticated user request %s", r.RemoteAddr)
+		http.Error(rw, "Unauthorized", http.StatusUnauthorized)
```

Comment:
Is `Unauthorized` the expected response, or should it be some form of Unauthenticated? Presumably the `NewUnauthenticatedHandler` function also handles writing the correct `metav1.Status` object back to the user?

Reply:
I think Unauthorized is correct.

All the handlers use the same error handler (this one) to determine how to handle the error. No k8s object json is sent back to the user if it fails at the proxy, not sure it should

Reply:
As an example of an 'unauthorized' response... https://35.195.25.157/api/v1/namespaces/default/pods

```
{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {
    
  },
  "status": "Failure",
  "message": "pods is forbidden: User \"system:anonymous\" cannot list resource \"pods\" in API group \"\" in the namespace \"default\": No policy matched.",
  "reason": "Forbidden",
  "details": {
    "kind": "pods"
  },
  "code": 403
}
```

It appears that you can't really ever be 'unauthenticated', you can only be assigned to the user `system:anonymous` (meaning there's no such thing as failed *authentication*)

Reply:
Yes indeed, that makes sense. I'm still not sure about the JSON response though, what if I'm using this for a non-k8s backend target. Would this format be what you would expect?

Reply:
How to handle non-k8s backends is definitely something that should be considered, but I'd say that if we are going to use this in front of a Kubernetes apiserver, there should be parity in behaviour else some clients may cease working/operate unexpectedly.

I don't think this is a net-change introduced by this PR though, so perhaps we should merge this first and open up an issue to discuss a plan going forward?